extensions/amp-story/0.1/animation.js

```diff
@@ -0,0 +1,627 @@
+  }
+
+  /** @private */
+  getRunners_() {
+    return dev().assert(this.runners_, 'Executed before applyFirstFrame');
+  }
+
+  /**
+   * @return {!Array<!Promise<!AnimationRunner>>}
+   * @private
+   */
+  getOrCreateRunners_() {
+    if (!this.runners_) {
+      this.runners_ = Array.prototype.map.call(
+          scopedQuerySelectorAll(this.root_, ANIMATABLE_ELEMENTS_SELECTOR),
+          el => this.createRunner_(el));
+    }
+    return dev().assert(this.runners_);
+  }
+
+  /**
+   * @param {!Element} el
+   * @return {!Promise<!AnimationRunner>}
+   */
+  createRunner_(el) {
+    const preset = this.getPreset_(el);
+    const animationDef = this.createAnimationDef(el, preset);
+
+    return new AnimationRunner(
+        this.root_,
+        animationDef,
+        dev().assert(this.builderPromise_),
+        this.vsync_,
+        this.timer_,
+        this.sequence_);
+  }
+
+  /**
+   * @param {!Element} el
+   * @param {!./animation-types.StoryAnimationPresetDef} preset
+   * @return {!./animation-types.StoryAnimationDef}
+   */
+  createAnimationDef(el, preset) {
+    const animationDef = {target: el, preset};
+
+    if (el.hasAttribute(ANIMATE_IN_DURATION_ATTRIBUTE_NAME)) {
+      animationDef.duration =
+          timeStrToMillis(el.getAttribute(ANIMATE_IN_DURATION_ATTRIBUTE_NAME));
+    }
+
+    if (el.hasAttribute(ANIMATE_IN_DELAY_ATTRIBUTE_NAME)) {
+      animationDef.delay =
+          timeStrToMillis(el.getAttribute(ANIMATE_IN_DELAY_ATTRIBUTE_NAME));
+    }
+
+    if (el.hasAttribute(ANIMATE_IN_AFTER_ATTRIBUTE_NAME)) {
+      const dependencyId = el.getAttribute(ANIMATE_IN_AFTER_ATTRIBUTE_NAME);
+
+      user().assertElement(
+          scopedQuerySelector(this.root_, `#${dependencyId}`),
+          `The attribute '${ANIMATE_IN_AFTER_ATTRIBUTE_NAME}' in tag ` +
+              `'${el.tagName}' is set to the invalid value ` +
+              `'${dependencyId}'. No children of parenting 'amp-story-page' ` +
+              `exist with id ${dependencyId}.`);
+
+      animationDef.startAfterId =
+          el.getAttribute(ANIMATE_IN_AFTER_ATTRIBUTE_NAME);
+    }
+
+    return animationDef;
+  }
+
+  /**
+   * @return {!Promise<!../../amp-animation/0.1/web-animations.Builder>}
+   * @private
+   */
+  createAnimationBuilderPromise_() {
+    return Services.extensionsFor(this.ampdoc_.win)
+        .loadExtension('amp-animation')
+        .then(() => Services.webAnimationServiceFor(this.ampdoc_))
+        .then(webAnimationService => webAnimationService.createBuilder());
+  }
+
+  /**
+   * @param {string} name
+   * @return {?./animation-types.StoryAnimationPresetDef}
+   */
+  getPreset_(el) {
+    const name = el.getAttribute(ANIMATE_IN_ATTRIBUTE_NAME);
+
+    return user().assert(
+        PRESETS[name],
+        'Invalid %s preset "%s" for element %s',
+        ANIMATE_IN_ATTRIBUTE_NAME,
+        name,
+        el);
+  }
+}
+
+
+/** Bus for animation sequencing. */
+class AnimationSequence {
```

Comment:
`signals.js` does something pretty similar FYI.

Reply:
Oh, that's cool. I'll look into whether the bus can be replaced with this.